src/GridElements.cpp

```diff
@@ -1788,6 +1788,82 @@ int BuildCoincidentNodeVector(
 	return nCoincidentNodes;
 }
 
+Real CalculateTriangleAreaQuadratureMethod(
+    Node node1, Node node2, Node node3
+) {
+
+    const int nOrder = 6;
```

Comment:
Why is this method introduced ? If you want to refactor, then you should also update the code in `CalculateFaceAreaQuadratureMethod` method to use `CalculateTriangleAreaQuadratureMethod` for each triangle. Else this is just repeated code and adds to maintenance burden

Reply:
There are other problems that should be fixed in amother PR; this is just for decomposition strategy; For example, the weights and gauss point locations should be static arrays, they should not be allocated every time. I will open an issue